nf_core/components/components_command.py

```diff
@@ -19,24 +20,34 @@ class ComponentCommand:
     Base class for the 'nf-core modules' and 'nf-core subworkflows' commands
     """
 
-    def __init__(self, component_type, dir, remote_url=None, branch=None, no_pull=False, hide_progress=False):
+    def __init__(
+        self,
+        component_type: str,
+        dir: str,
+        remote_url: str = None,
+        branch: str = None,
+        no_pull: bool = False,
+        hide_progress: bool = False,
+    ) -> None:
         """
         Initialise the ComponentClass object
         """
         self.component_type = component_type
         self.dir = dir
-        self.modules_repo = ModulesRepo(remote_url, branch, no_pull, hide_progress)
+        self.modules_repo: ModulesRepo = ModulesRepo(remote_url, branch, no_pull, hide_progress)
         self.hide_progress = hide_progress
         self._configure_repo_and_paths()
 
-    def _configure_repo_and_paths(self, nf_dir_req=True):
+    def _configure_repo_and_paths(self, nf_dir_req: Optional[bool] = True) -> None:
         """
         Determine the repo type and set some default paths.
         If this is a modules repo, determine the org_path too.
 
         Args:
             nf_dir_req (bool, optional): Whether this command requires being run in the nf-core modules repo or a nf-core pipeline repository. Defaults to True.
         """
+        self.dir: Union[str, Path]
+        self.repo_type: Optional[str]
```

Comment:
`self.dir` is already defined as `str` in the `__init__` so this seems to conflict with that. But more than that it can also be `None`, hence the below code with `if self.dir` so I think the real type is probably `Optional[Union[str, Path]]`? 

Is this the right place for these declarations though? And should `self.org` not be included if it is?

Reply:
sorry I missed these things before accepting, this makes sense! 😄

Reply:
I agree with @awgymer in creating custom types. For example, in the [PR #2214 ](https://github.com/nf-core/tools/pull/2214) we have 
[an instance where a custom type is a good idea](https://github.com/nf-core/tools/blob/bba34d20f8c20acfef312c6f1a7213d596d0b274/nf_core/components/list.py#L113).

Reply:
My suggestion here is to remove these two lines :) 
```suggestion
```

Reply:
It would be good if 
```python
self.dir, self.repo_type, self.org = get_repo_info(self.dir, use_prompt=nf_dir_req)
```
returns satisfactory types, so explicit declarations above the function call are not necessary.